### PR TITLE
fix: add fallbacks for VIP-specific code

### PR DESCRIPTION
**Pull request by sjinks**
This PR adds replacements for:
* `\Automattic\VIP\Utils\Alerts` (partial implementation only for the used methods);
* `\Automattic\VIP\Logstash\Logger` (partial implementation);
* `\Automattic\VIP\Logstash\log2logstash()` (replaced with `Logger::log2logstash()`);
* `\Automattic\VIP\Search\Search::should_load_new_ep()` (assume it always returns `true`).

References to `WPCOM_VIP_CLI_Command` are replaced with `WP_CLI_Command` (we don't seem to use any of `WPCOM_VIP_CLI_Command`'s features).

Calls to `vip_reset_local_object_cache()` and `vip_safe_wp_remote_request()` are guarded with `function_exists()`.

A couple of other bugs fixed, see Slack.

### CODE REVIEW

File: src/includes/classes/class-search.php [hunk 1]
```diff
@@ -1707,7 +1689,7 @@ public function filter__ep_post_mapping( $mapping ) {
 	public function filter__ep_user_mapping( $mapping ) {
 		$users_count = count_users();
 
-		if ( isset( $users_count->total_users ) && ( $users_count->total_users > self::USER_SHARD_THRESHOLD ) ) {
+		if ( isset( $users_count['total_users'] ) && ( $users_count['total_users'] > self::USER_SHARD_THRESHOLD ) ) {
```
Comment (sjinks) [inline src/includes/classes/class-search.php:1692]: See Slack why.

File: src/includes/classes/class-versioning.php [hunk 2]
```diff
@@ -991,7 +998,7 @@ public function maybe_self_heal() {
 	}
 
 	public function alert_for_index_self_healing( string $slug ) {
-		if ( ! isset( $this->alert ) ) {
+		if ( ! isset( $this->alerts ) ) {
```
Comment (sjinks) [inline src/includes/classes/class-versioning.php:1001]: See Slack why.

File: src/includes/classes/class-versioning.php [hunk 3]
```diff
@@ -1006,7 +1013,7 @@ public function alert_for_index_self_healing( string $slug ) {
 	}
 
 	public function alert_for_index_self_healing_failed( string $slug ) {
-		if ( ! isset( $this->alert ) ) {
+		if ( ! isset( $this->alerts ) ) {
```
Comment (sjinks) [inline src/includes/classes/class-versioning.php:1016]: See Slack why.

File: src/includes/classes/commands/class-versioncommand.php [hunk 4]
```diff
@@ -123,6 +123,7 @@ public function get( $args, $assoc_args ) {
 				restore_current_blog();
 
 				if ( is_wp_error( $version ) ) {
+					/** @var WP_Error $version */
```
Comment (sjinks) [inline src/includes/classes/commands/class-versioncommand.php:126]: Here and below: this reduces noise from Intelephense when it cannot infer the correct variable type.

File: src/includes/classes/class-search.php [hunk 5]
```diff
@@ -970,7 +952,7 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 		 * Skip vip_safe_wp_remote_request for non-query (search) requests
 		 * Any timeouts happening in non-search/non-query context shouldn't count towards the request disabling threshold.
 		 */
-		if ( 'query' === $type ) {
+		if ( 'query' === $type && function_exists( 'vip_safe_wp_remote_request' ) ) {
```
Comment (sjinks) [inline src/includes/classes/class-search.php:955]: @rebeccahum is this check safe?

File: src/includes/classes/commands/class-documentcommand.php [hunk 6]
```diff
@@ -3,15 +3,14 @@
 namespace Automattic\VIP\Search\Commands;
 
 use WP_CLI;
-
-require_once __DIR__ . '/../../../../vip-helpers/vip-wp-cli.php';
+use WP_CLI_Command;
```
Comment (rebeccahum) [inline src/includes/classes/commands/class-documentcommand.php:6]: For the WP_CLI_Command, checked the code to make sure it doesn't make any calls to WPCOM_VIP_CLI_Command ✅

File: src/includes/classes/class-search.php (hunk 5 above)
Comment (rebeccahum) [reply src/includes/classes/class-search.php:955]: yes, should be fine!

**Review by rebeccahum — APPROVED** [review]

**Pull request merged**